apollo-router/src/plugins/connectors/http_json_transport.rs

```diff
@@ -114,86 +117,36 @@ pub(crate) fn make_request(
 }
 
 fn make_uri(
-    transport: &HttpJsonTransport,
+    source_url: Option<&Url>,
+    template: &URLTemplate,
     inputs: &IndexMap<String, Value>,
 ) -> Result<Url, HttpJsonTransportError> {
     let flat_inputs = flatten_keys(inputs);
-    let generated = transport
-        .connect_template
-        .generate(&flat_inputs)
-        .map_err(HttpJsonTransportError::TemplateGenerationError)?;
-    if let Some(source_url) = transport.source_url.as_ref() {
-        append_path(
-            Url::parse(source_url).map_err(HttpJsonTransportError::InvalidUrl)?,
-            &generated,
-        )
-    } else {
-        Url::parse(&generated).map_err(HttpJsonTransportError::InvalidUrl)
-    }
-}
+    let mut url = source_url
+        .or(template.base.as_ref())
+        .ok_or(HttpJsonTransportError::NoBaseUrl)?
+        .clone();
```

Comment:
In theory we only have to `.clone()` if we're going to add path &| query params, but I figure that's the vast majority of the time